src/Uno.UI/Controls/NativeCommandBarPresenter.iOS.cs

```diff
@@ -69,33 +71,47 @@ void OrientationChanged(DisplayInformation displayInformation, object args)
 
 		private double GetCommandBarHeight()
 		{
-			// For phones with a screen size of 4.7 inches or less, the navigation bar size is different in portrait and landscape mode
-			// Navigation bar height depending of device : https://kapeli.com/cheat_sheets/iOS_Design.docset/Contents/Resources/Documents/index
-			// Devices screen specifications : https://www.idev101.com/code/User_Interface/sizes.html
-			var bounds = UIScreen.MainScreen.Bounds;
-			var height = Math.Max(bounds.Width, bounds.Height);
-			var width = Math.Min(bounds.Width, bounds.Height);
-			var isPhoneSmallScreen = height <= 667 && width <= 375;
-			
-			if (isPhoneSmallScreen && UIDevice.CurrentDevice.UserInterfaceIdiom == UIUserInterfaceIdiom.Phone)
+			if (UIDevice.CurrentDevice.UserInterfaceIdiom == UIUserInterfaceIdiom.Phone)
 			{
-				var orientation = DisplayInformation.GetForCurrentView().CurrentOrientation;
+				// For phones with a screen size of 4.7 inches or less, the navigation bar size is different in portrait and landscape mode
+				// Navigation bar height depending of device : https://kapeli.com/cheat_sheets/iOS_Design.docset/Contents/Resources/Documents/index
+				// Devices screen specifications : https://www.idev101.com/code/User_Interface/sizes.html
+				var bounds = UIScreen.MainScreen.Bounds;
+				var height = Math.Max(bounds.Width, bounds.Height);
+				var width = Math.Min(bounds.Width, bounds.Height);
+				var isPhoneSmallScreen = height <= 667 && width <= 375;
 
-				switch (orientation)
+				if (isPhoneSmallScreen || UIDevice.CurrentDevice.CheckSystemVersion(11, 0))
 				{
-					case DisplayOrientations.Landscape:
-					case DisplayOrientations.LandscapeFlipped:
-						return _landscapeSmallPhoneCommandBarHeight;
-					case DisplayOrientations.Portrait:
-					case DisplayOrientations.PortraitFlipped:
-					case DisplayOrientations.None:
-					default:
-						return _defaultCommandBarHeight;
+					var orientation = DisplayInformation.GetForCurrentView().CurrentOrientation;
+
+					switch (orientation)
+					{
+						case DisplayOrientations.Landscape:
+						case DisplayOrientations.LandscapeFlipped:
+							return _landscapeSmallPhoneCommandBarHeight;
+						case DisplayOrientations.Portrait:
+						case DisplayOrientations.PortraitFlipped:
+						case DisplayOrientations.None:
+						default:
+							return _defaultCommandBarHeight;
+					}
+				}
+				else
+				{
+					return _defaultCommandBarHeight;
 				}
 			}
 			else
 			{
-				return _defaultCommandBarHeight;
+				if (UIDevice.CurrentDevice.CheckSystemVersion(12, 0))
+				{
+					return _iPad12CommandBarHeight;
```

Comment:
What about if it's a page with a Large Title, won't it be bigger?

Is there no way to extract the real height at runtime? Relying on hardcoded values like this seems to be brittle.

Reply:
Large titles are not supported by the CommandBar.

And I've tried [this solution](https://stackoverflow.com/questions/7312059/programmatically-get-height-of-navigation-bar) to get height on runtime but the UINavigationBar is always null in our case.